rotkehlchen/chain/ethereum/modules/convex/decoder.py

```diff
@@ -71,19 +75,42 @@ def _decode_convex_events(self, context: DecoderContext) -> DecodingOutput:
                     event.counterparty = CPT_CONVEX
                     if context.tx_log.address in CONVEX_POOLS:
                         event.notes = f'Deposit {event.balance.amount} {crypto_asset.symbol} into convex {CONVEX_POOLS[context.tx_log.address]} pool'  # noqa: E501
+                        event.product = EvmProduct.CONVEX_GAUGE
                     else:
                         event.notes = f'Deposit {event.balance.amount} {crypto_asset.symbol} into convex'  # noqa: E501
+                        if (
+                            isinstance(crypto_asset, EvmToken) and
+                            crypto_asset.protocol == CURVE_POOL_PROTOCOL
+                        ):
+                            event.product = EvmProduct.CONVEX_GAUGE
+
+                    # in this case store information about the gauge in the extra details to use
+                    # it during balances queries
+                    for log_event in context.all_logs:
+                        if log_event.topics[0] == DEPOSIT_EVENT:
+                            deposit_amount_raw = hex_or_bytes_to_int(context.tx_log.data[0:32])
+                            staking_address = hex_or_bytes_to_address(log_event.topics[1])
+                            if deposit_amount_raw == amount_raw and staking_address == event.location_label:  # noqa: E501
+                                event.extra_data = {'gauge_address': log_event.address}
+                                break
+
             elif (
                 event.event_type == HistoryEventType.RECEIVE and
                 event.event_subtype == HistoryEventSubType.NONE
             ):
                 if context.tx_log.topics[0] in WITHDRAWAL_TOPICS:
                     event.event_type = HistoryEventType.WITHDRAWAL
+                    event.counterparty = CPT_CONVEX
                     if context.tx_log.address in CONVEX_POOLS:
                         event.notes = f'Withdraw {event.balance.amount} {crypto_asset.symbol} from convex {CONVEX_POOLS[context.tx_log.address]} pool'  # noqa: E501
+                        event.product = EvmProduct.CONVEX_GAUGE
                     else:
                         event.notes = f'Withdraw {event.balance.amount} {crypto_asset.symbol} from convex'  # noqa: E501
-                    event.counterparty = CPT_CONVEX
+                        if (
+                            isinstance(crypto_asset, EvmToken) and
+                            crypto_asset.protocol == CURVE_POOL_PROTOCOL
+                        ):
+                            event.product = EvmProduct.CONVEX_GAUGE
```

Comment:
I would adjust notes here

Reply:
Not even a pool address? @yabirgb that sounds weird. Where from do you withdraw this amount? What is this `else` clause?

Reply:
In the case of the deposit I had in the tests that is for `steCRV` (eth/steth) pool it falls under the else statement. The reason is because there was a migration at some point irc and the old transactions are recorded with the old pools. I guess that we should keep this history (probably by looking at older versions of the registry) and then this would fall under the if statement. There is also CVX deposits in this part of the if/else statement

Reply:
Then you should really add this explanation at the else. And yes we could in the future look for older pools or just hard-code them?